### PR TITLE
workaround to must-gather sig

### PR DESCRIPTION
**Pull request by eranco74**
currently the signature show up in cases that we know that the must-gather can't be collected.
With this fix triage tickets will get this signature only if must-gather logs should've been collected - the bootstrap node went to reboot. In case of SNO the signature will be added only if we the controller start running post reboot.

### CODE REVIEW

File: tools/add_triage_signature.py [hunk 1]
```diff
@@ -1393,8 +1393,16 @@ def __init__(self, *args, **kwargs):
 
     def _process_ticket(self, url, issue_key):
         md = get_metadata_json(url)
-        cluster_id = md["cluster"]["id"]
+        cluster_hosts = md["cluster"]["hosts"]
+        bootstrap_node = [host for host in cluster_hosts if host["bootstrap"]][0]
+        eligible_bootstrap_stages = ["Rebooting", "Configuring", "Joined", "Done"]
```
Comment (osherdp) [inline tools/add_triage_signature.py:1398]: also ``Failed``, no?

File: tools/add_triage_signature.py [hunk 2]
```diff
@@ -1393,8 +1393,16 @@ def __init__(self, *args, **kwargs):
...
+        if len(cluster_hosts) <= 1:
+            # in SNO when the bootstrap node goes to reboot the controller is still not running
+            eligible_bootstrap_stages.remove("Rebooting")
```
Comment (osherdp) [inline tools/add_triage_signature.py:1401]: append blank line here, I'm choking here from all those lines XD

File: tools/add_triage_signature.py [hunk 3]
```diff
@@ -1393,8 +1393,16 @@ def __init__(self, *args, **kwargs):
...
+        if bootstrap_node["progress"]["current_stage"] not in eligible_bootstrap_stages:
+            return
```
Comment (osherdp) [inline tools/add_triage_signature.py:1403]: here too

**Review by osherdp — CHANGES_REQUESTED** [review]
looks good. styling and a question

File: tools/add_triage_signature.py (hunk 1 above)
Comment (eranco74) [reply tools/add_triage_signature.py:1398]: Nope, this is progress stage, not state

Comment (eranco74) [reply tools/add_triage_signature.py:1399]: Bootstrap progress stages are:
         "Starting installation",
          "Installing",
          "Writing image to disk",
          "Waiting for control plane",
          "Waiting for bootkube",
          "Waiting for controller",
          "Rebooting",
          "Configuring",
          "Joined",
          "Done"

**Review by osherdp — APPROVED** [review]

**Pull request merged**